operator-common/src/main/java/io/strimzi/operator/common/model/Labels.java

```diff
@@ -196,68 +215,74 @@ private Labels with(String label, String value) {
      */
-    public Labels withCluster(String cluster) {
+    public Labels withStrimziCluster(String cluster) {
         return with(STRIMZI_CLUSTER_LABEL, cluster);
     }
 
     /**
      * The same labels as this instance, but with the application name {@code strimzi} for the {@code app.kubernetes.io/name} key.
+     *
+     * @param name The kubernetes name to add.
      * @return A new instance with the given kubernetes application name added.
      */
-    public Labels withKubernetesName() {
-        return with(Labels.KUBERNETES_NAME_LABEL, Labels.KUBERNETES_NAME);
+    public Labels withKubernetesName(String name) {
+        return with(Labels.KUBERNETES_NAME_LABEL, name);
     }
 
     /**
      * The same labels as this instance, but with the given {@code instance} for the {@code app.kubernetes.io/instance} key.
-     * @param instance The instance to add.
+     *
+     * @param instanceName The instance to add.
      * @return A new instance with the given kubernetes application instance added.
      */
-    public Labels withKubernetesInstance(String instance) {
-        return with(Labels.KUBERNETES_INSTANCE_LABEL, getOrValidInstanceLabelValue(instance));
+    public Labels withKubernetesInstance(String instanceName) {
+        return with(Labels.KUBERNETES_INSTANCE_LABEL, getOrValidInstanceLabelValue(instanceName));
     }
 
     /**
      * The same labels as this instance, but with the given {@code part-of} for the {@code app.kubernetes.io/part-of} key.
-     * @param partof The partof label to add.
+     *
+     * @param instanceName The instance used to generate the unique label to add composed with the application name.
      * @return A new instance with the given kubernetes application part-of label added.
      */
-    public Labels withKubernetesPartOf(String partof) {
-        return with(Labels.KUBERNETES_PART_OF_LABEL, getOrValidInstanceLabelValue(partof));
+    public Labels withKubernetesPartOf(String instanceName) {
+        return with(Labels.KUBERNETES_PART_OF_LABEL, getOrValidInstanceLabelValue(APPLICATION_NAME + "-" + instanceName));
     }
 
     /**
      * Validates the instance name and if needed modifies it to make it a valid Label value:
-     *   - (([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?
-     *   - 63 characters max
+     * - (([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?
+     * - 63 characters max
      * This method is written to handle instance names which are valid resource names, since they are derived from a
      * custom resource. It does not modify arbitrary names as label values.
      *
-     *
      * @param instance Theoriginal name of the instance
      * @return Either the original instance name or a modified version to match label value criteria
      */
-    /*test*/ static String getOrValidInstanceLabelValue(String instance) {
-        if (instance == null)   {
+    /*test*/
+    static String getOrValidInstanceLabelValue(String instance) {
```

Comment:
I think we normally have this on single line:

```suggestion
    /*test*/ static String getOrValidInstanceLabelValue(String instance) {
```